apps/create-vesting/src/app/page.tsx

```diff
@@ -72,12 +99,34 @@ export default function Page() {
   const onSubmitCallback = useCallback(
     async (data: CreateVestingFormData) => {
       if (!context?.account || !token || !vestingEscrowFactoryAddress) return;
+
+      // Validate ENS name and get address
+      clearErrors("recipient");
+      let address: string;
+      try {
+        address = await validateRecipient(data.recipient);
+      } catch (error) {
+        if (error instanceof Error) {
+          setError("recipient", {
+            type: "manual",
+            message: error.message,
+          });
+        } else {
+          setError("recipient", {
+            type: "manual",
+            message: "Couldn't verify ENS name",
+          });
+        }
+        return;
+      }
```

Comment:
I think that this should be made on the validation form stage not on submission.

Reply:
The point is that I need the address in this section of the code, so I can pass the recipient address, not the ens name, to the hooks. If I add this to zod validation we would be fetching this twice